client-react/src/pages/app/app-settings/ApplicationSettings/AppSettingAddEdit.tsx

```diff
@@ -176,8 +184,8 @@ const AppSettingAddEdit: React.SFC<AppSettingAddEditProps> = props => {
           secondaryButton={actionBarSecondaryButtonProps}
         />
       </form>
-      {isAppSettingReferenceVisible() && isValidKeyVaultReference() && !!currentAppSettingReference && (
-        <KeyVaultReferenceComponent resourceId={site.id} appSettingReference={currentAppSettingReference} />
+      {isAppSettingReferenceVisible() && isValidKeyVaultReference() && isValidAppConfigReference() && !!currentAppSettingReference && (
```

Comment:
shouldn't this condition be - 

```suggestion
      {isAppSettingReferenceVisible()  && !!currentAppSettingReference && (isValidKeyVaultReference() || isValidAppConfigReference()) && (
```

Reply:
it cannot be both a valid keyvault and a validAppConfig at the same time, correct? it should be either of the one

Reply:
ya just corrected this